library/options.inc.php

```diff
@@ -122,7 +122,33 @@ function generate_select_list($tag_name, $list_id, $currvalue, $title, $empty_na
 	$selectEmptyName = xlt($empty_name);
 	if ($empty_name)
 		$s .= "<option value=''>" . $selectEmptyName . "</option>";
-	$lres = sqlStatement("SELECT * FROM list_options WHERE list_id = ? AND activity=1 ORDER BY seq, title", array($list_id));
+
+        // List order depends on language translation options.
+        //  (Note we do not need to worry about the list order in the algorithm
+        //   after the below code block since that is where searches for exceptions
+        //   are done which include inactive items or items from a backup
+        //   list; note these will always be shown at the bottom of the list no matter the
+        //   chosen order.)
+        $lang_id = empty($_SESSION['language_choice']) ? '1' : $_SESSION['language_choice'];
+        if ($GLOBALS['gb_how_sort_list'] == '0') {
+          //sort by seq
+          $lres = sqlStatement("SELECT * FROM list_options WHERE list_id = ? AND activity=1 ORDER BY seq, title", array($list_id));
```

Comment:
The secondary sort is by title so you should use the logic below instead, only with ORDER BY being different.  It's not unusual to have many items with the same sequence number as a way to effectively sort by title.